### PR TITLE
nix-init: refactor with nixfmt-rfc-style, passthru.tests.version and removing darwin sdk

### PR DESCRIPTION
**Pull request by kachick**
## Things done

```console
> nix-build --attr pkgs.nix-init.passthru.tests     
this derivation will be built:
  /nix/store/54850gkq3dm8rmva6d14cjlw2gkw0g0d-nix-init-0.3.2-test-version.drv
this path will be fetched (6.46 MiB download, 18.98 MiB unpacked):
  /nix/store/d371fzmyvnssxkgf1r1vqg47fd8y63i0-nix-init-0.3.2
copying path '/nix/store/d371fzmyvnssxkgf1r1vqg47fd8y63i0-nix-init-0.3.2' from 'https://cache.nixos.org'...
building '/nix/store/54850gkq3dm8rmva6d14cjlw2gkw0g0d-nix-init-0.3.2-test-version.drv'...
nix-init 0.3.2
/nix/store/j6ln6ysgqv4kyliq7m40llrglp5layqg-nix-init-0.3.2-test-version
```

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kachick** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356985`

---
### `x86_64-linux`
## `nixpkgs-review` result

---
### `x86_64-darwin`

---
### `aarch64-darwin`
## `nixpkgs-review` result

File: pkgs/by-name/ni/nix-init/package.nix [hunk 1]
```diff
@@ -45,18 +48,21 @@ rustPlatform.buildRustPackage rec {
...
+      openssl
+      zlib
+      zstd
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      darwin.apple_sdk.frameworks.Security
+    ]
+    ++ lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isx86_64) [
+      darwin.apple_sdk.frameworks.CoreFoundation
+    ];
```
Comment (uncenter) [inline pkgs/by-name/ni/nix-init/package.nix:65]: Fwiw these should no longer be required now that the darwin SDKs are included by default, if you want to try removing these.

**Comment by uncenter** [comment]
(You could probably amend that last commit into `nix-init: remove unnecessary darwin sdk` commit if you want :p)

**Comment by kachick** [comment]
Thank you for reviewing! Now rebuilding the nixpkgs-review again, I'll post the result after finish :pray:

**Comment by uncenter** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356985`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>nix-init</li>
  </ul>
</details>

**Review by uncenter — APPROVED** [review]

**Comment by Aleksanaa** [comment]
@ofborg build nix-init

**Comment by kachick** [comment]
Thanks for your reviews!

**Pull request merged**